File: fs/BeginnersSelection/06_CardGameForTwo.fs
// https://atcoder.jp/contests/abs/tasks/abc088_b

stdin.ReadLine()
|> ignore
|> fun () -> stdin.ReadLine()
|> fun x -> x.Split(' ')
|> Array.map int
|> Array.sortWith (fun a b -> b - a)
|> Array.mapi (fun i x -> (i, x))
|> Array.fold (fun (a, b) (i, x) ->
  (
    if i % 2 = 0
    then (a + x, b)
    else (a, b + x)
  )
) (0, 0)
|> fun (a, b) -> a - b
|> printfn "%d"

// ----------

stdin.ReadLine() |> ignore |> fun () -> stdin.ReadLine() |> fun x -> x.Split(' ') |> Array.map int |> Array.sortWith (fun a b -> b - a) |> Array.mapi (fun i x -> (i, x)) |> Array.fold (fun (a, b) (i, x) -> if i % 2 = 0 then (a + x, b) else (a, b + x) ) (0, 0) |> fun (a, b) -> a - b |> printfn "%d"

// https://atcoder.jp/contests/abs/submissions/9414786

// ----------

// mapi はあるのに foldi はない。。

File: ex/BeginnersSelection/06_CardGameForTwo.ex
# https://atcoder.jp/contests/abs/tasks/abc088_b

defmodule Main, do:
  def main(), do:
    [IO.read(:line), IO.read(:line)]
    |> Enum.at(1)
    |> String.trim
    |> String.split(" ")
    |> Enum.map(&String.to_integer/1)
    |> Enum.sort(& &1 > &2)
    |> Enum.reduce([0, 0, 0], fn x, t -> apply(fn a, b, i ->
      if rem(i, 2) === 0,
        do:   [a + x, b, i + 1],
        else: [a, b + x, i + 1]
    end, t) end)
    |> (& Enum.at(&1, 0) - Enum.at(&1, 1)).()
    |> IO.puts

# https://atcoder.jp/contests/abs/submissions/21758902
